src/Hcal/HcalRecProducer.cxx

```diff
@@ -330,7 +351,7 @@ void HcalRecProducer::produce(framework::Event& event) {
     recHit.setZPos(position.Z());
     recHit.setPE(PEs);
     recHit.setMinPE(minPEs);
-    recHit.setAmplitude(amplT_posend);
+    recHit.setAmplitude(amplT);
     recHit.setEnergy(energy_deposited);
```

Comment:
Use `reconstructed_energy` instead of `energy_deposited` so future changes to the sampling fraction are captured.

Reply:
great suggestion